# **TITANIC SURVIVOR ANALYSIS**

In [15]:
import pandas as pd
import numpy as np

In [16]:
df = pd.DataFrame(pd.read_csv('/content/train (1).csv'))
df.head()

PassengerId  Survived  Pclass  ...     Fare Cabin  Embarked
0            1         0       3  ...   7.2500   NaN         S
1            2         1       1  ...  71.2833   C85         C
2            3         1       3  ...   7.9250   NaN         S
3            4         1       1  ...  53.1000  C123         S
4            5         0       3  ...   8.0500   NaN         S

[5 rows x 12 columns]

In [17]:
df.shape

(891, 12)

This shape function is the analysis that there are 891 rows and 12 columns in the DataFrame.

In [18]:
df.isnull().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

The aforementioned analysis is done in order to handle NULL values that might be a part of our DataFrame.

In [19]:
drop_col = df.isnull().sum()[df.isnull().sum()>(35/100 * df.shape[0])]
drop_col

Cabin    687
dtype: int64

In the Above code snippet we are actually separating the columns from our DataFrame which have more than 35 % of their values as NULL.

In [33]:
drop_col.index

Index(['Cabin'], dtype='object')

In [40]:
df.fillna(df.mean(), inplace=True)
df.isnull().sum()

PassengerId    0
Survived       0
Pclass         0
Name           0
Sex            0
Age            0
SibSp          0
Parch          0
Ticket         0
Fare           0
Embarked       2
dtype: int64

Embarked Column here contains String values so mean was not able to be calculated hence we are still seeing that it has two null values.

In [41]:
df['Embarked'].describe()

count     889
unique      3
top         S
freq      644
Name: Embarked, dtype: object

In [43]:
df['Embarked'].fillna("S", inplace=True)
df.isnull().sum()

PassengerId    0
Survived       0
Pclass         0
Name           0
Sex            0
Age            0
SibSp          0
Parch          0
Ticket         0
Fare           0
Embarked       0
dtype: int64

Finally All the NULL values have been dealt with.

In [44]:
df.corr()

PassengerId  Survived    Pclass  ...     SibSp     Parch      Fare
PassengerId     1.000000 -0.005007 -0.035144  ... -0.057527 -0.001652  0.012658
Survived       -0.005007  1.000000 -0.338481  ... -0.035322  0.081629  0.257307
Pclass         -0.035144 -0.338481  1.000000  ...  0.083081  0.018443 -0.549500
Age             0.033207 -0.069809 -0.331339  ... -0.232625 -0.179191  0.091566
SibSp          -0.057527 -0.035322  0.083081  ...  1.000000  0.414838  0.159651
Parch          -0.001652  0.081629  0.018443  ...  0.414838  1.000000  0.216225
Fare            0.012658  0.257307 -0.549500  ...  0.159651  0.216225  1.000000

[7 rows x 7 columns]

In [45]:
df['FamilyMan'] = df['SibSp'] + df['Parch']
df.drop(['SibSp', 'Parch'], axis=1, inplace=True)
df.corr()


PassengerId  Survived    Pclass       Age      Fare  FamilyMan
PassengerId     1.000000 -0.005007 -0.035144  0.033207  0.012658  -0.040143
Survived       -0.005007  1.000000 -0.338481 -0.069809  0.257307   0.016639
Pclass         -0.035144 -0.338481  1.000000 -0.331339 -0.549500   0.065997
Age             0.033207 -0.069809 -0.331339  1.000000  0.091566  -0.248512
Fare            0.012658  0.257307 -0.549500  0.091566  1.000000   0.217138
FamilyMan      -0.040143  0.016639  0.065997 -0.248512  0.217138   1.000000

Above, we created a new Column 'FamilyMan' by combining the two Columns 'SibSp' and 'Parch'.

Quick Conclusion: THe Survival Rate is not that greatly influenced by the FamilyMan i.e. the Survival Rate is irrespective of whether the Family is big or small.

In [46]:
df['SinglePerson'] = [0 if df['FamilyMan'][i]>0 else 1 for i in df.index]
df.head()

PassengerId  Survived  Pclass  ... Embarked FamilyMan  SinglePerson
0            1         0       3  ...        S         1             0
1            2         1       1  ...        C         1             0
2            3         1       3  ...        S         0             1
3            4         1       1  ...        S         1             0
4            5         0       3  ...        S         0             1

[5 rows x 11 columns]

Here 0 indicates False and 1 indicates True.

In [47]:
df.groupby(['SinglePerson'])['Survived'].mean()

SinglePerson
0    0.505650
1    0.303538
Name: Survived, dtype: float64

Well, the Above Data Analysis concludes that a Single Person or one is travelling Alone has lesser or fewer chance of Survival as a FamilyMan who was able to affor the ticket for his/her family shows that he/she would have been Rich and we all know how the Rich gets prioritised over all.

In [48]:
df[['SinglePerson', 'Fare']].corr()

SinglePerson      Fare
SinglePerson      1.000000 -0.271832
Fare             -0.271832  1.000000

Quick Conclusion: For someone not a SinglePerson, the chance of the Fare being Higher is High.

In [49]:
df['Sex'] = [0 if df['Sex'][i]=='male' else 1 for i in df.index] 
#Note: 0 is for Male and 1 is for FeMale
df.groupby(['Sex'])['Survived'].mean()

Sex
0    0.188908
1    0.742038
Name: Survived, dtype: float64

It is clearly visible that Female Passengers have more chance of survival than the Male ones. It seems that Female Passengers were prioritised over the Male Passengers.

In [50]:
df.groupby(['Embarked'])['Survived'].mean()

Embarked
C    0.553571
Q    0.389610
S    0.339009
Name: Survived, dtype: float64

# CONCLUSION:


1.   The Priority of Escape from the Sinking Ship was given to Female  Passengers over Male ones.   
2.   The Harsh Reality of the World, The Rich have higher survival rate than the others.
3.   Passengers travelling with their Families have higher rate of Survival.
4.   Passengers who boarded the Ship at Cherbourg, survived more in proprtions than the other wither they were very lucky or it was thier destiny.



